.github/workflows/dist.yml

```diff
@@ -0,0 +1,50 @@
+on:
+  release:
+    types: [published]
+
+name: Generate Dist Archive
+
+jobs:
+  dist:
+    runs-on: ubuntu-22.04
+    container: fedora:38
+    permissions:
+        contents: write # needed for uploading release artifact
+    steps:
+      - name: Register archive name
+        id: register_archive_name
+        shell: bash
```

Comment:
I don't think that this is needed here. It's needed in the windows installer workflow because the default shell is MSYS2 bash there, which can't write to `GITHUB_OUTPUT` properly.

Reply:
I see, thanks. Btw it is kind of unpleasant that we can only "match" to the tarball file naming, but I don't think there is a way to set the archive name by us in meson. So this workflow actually only works if the tag matches the version in meson

Reply:
Hm, maybe we can at least retrieve it from the file system instead of hardcoding it? We should be able to write the path of files that match a glob to the step output, and then use that in the following steps.

Reply:
After running `meson dist`, we could use the output of `ls _mesonbuild/meson-dist/*.tar.xz` and `ls _mesonbuild/meson-dist/*.tar.xz.sha256sum` to `$GITHUB_OUTPUT` (or stricter if there are multiple files with those extensions).

Reply:
Thats true, there shouldn't be any files in there, unless we would start using caching. I refactored it like you suggested